ui/src/test/java/org/pentaho/di/ui/trans/steps/tableoutput/TableOutputDialogTest.java

```diff
@@ -66,4 +79,40 @@ private RowMetaInterface createRowMeta( String[] args, boolean hasEmptyFields )
     }
     return result;
   }
+
+  private void isConnectionSupportedTest( boolean supported ) {
+
+    TableOutputDialog dialog = mock( TableOutputDialog.class );
+
+    TransMeta transMeta = mock( TransMeta.class );
+    DatabaseMeta dbMeta = mock( DatabaseMeta.class );
+    TextVar text = mock( TextVar.class );
+    CCombo combo = mock( CCombo.class );
+    DatabaseInterface dbInterface = mock( DatabaseInterface.class );
+
+    Whitebox.setInternalState( dialog, "wTable", text );
+    Whitebox.setInternalState( dialog, "wConnection", combo );
+    Whitebox.setInternalState( dialog, "transMeta", transMeta );
+
+    when( text.getText() ).thenReturn( "someTable" );
+    when( combo.getText() ).thenReturn( "someConnection" );
+    when( transMeta.findDatabase( anyString() ) ).thenReturn( dbMeta );
+    when( dbMeta.getDatabaseInterface() ).thenReturn( dbInterface );
+
+    doNothing().when( dialog ).showUnsupportedConnectionMessageBox( dbInterface );
+    doCallRealMethod().when( dialog ).isConnectionSupported();
+
+    //Check that if the db interface does not support standard output then showUnsupportedConnection is called
+    when( dbInterface.supportsStandardTableOutput() ).thenReturn( supported );
+    dialog.isConnectionSupported();
+    verify( dialog, times( !supported ? 1 : 0 ) ).showUnsupportedConnectionMessageBox( dbInterface );
+  }
+
+  @Test
+  public void isConnectionSupportedTest() {
+
+    isConnectionSupportedTest( false );

```

Comment:
Write one test for `true` and another for `false`. This way, when one fails, you'll not which code path is faulty.